include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
  *
- * @return An IoT Error Type defining successful/failed unsubscribe call
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_unsubscribe] */
 IoT_Error_t aws_iot_mqtt_unsubscribe(AWS_IoT_Client *pClient, const char *pTopicFilter, uint16_t topicFilterLen);
+/* @[declare_mqtt_unsubscribe] */
 
 /**
- * @brief Disconnect an MQTT Connection
+ * @brief Disconnect an MQTT session.
  *
- * Called to send a disconnect message to the broker.
+ * This function sends the MQTT DISCONNECT packet, which closes the MQTT session
+ * between the client and server. After this function returns, the MQTT client
+ * context should be either freed with @ref mqtt_function_free or reopened with
+ * @ref mqtt_function_connect.
  *
- * @param pClient Reference to the IoT Client
+ * @param[in] pClient MQTT client context
  *
- * @return An IoT Error Type defining successful/failed send of the disconnect control packet.
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_disconnect] */
 IoT_Error_t aws_iot_mqtt_disconnect(AWS_IoT_Client *pClient);
+/* @[declare_mqtt_disconnect] */
 
 /**
- * @brief Yield to the MQTT client
+ * @brief Provide time for the MQTT client to process events.
  *
- * Called to yield the current thread to the underlying MQTT client.  This time is used by
- * the MQTT client to manage PING requests to monitor the health of the TCP connection as
- * well as periodically check the socket receive buffer for subscribe messages.  Yield()
- * must be called at a rate faster than the keepalive interval.  It must also be called
- * at a rate faster than the incoming message rate as this is the only way the client receives
- * processing time to manage incoming messages.
+ * This function processes the following events:
+ * - Incoming messages from the server <br>
+ * Topic filter subscriptions registered with @ref mqtt_function_subscribe instruct
+ * the server to send matching messages to this client. These messages are received
+ * by this function, which then calls the appropriate message handler. This function
+ * must be called at a rate faster than the incoming messages, as it is the only way
+ * the client receives processing time to manage incoming messages.
+ * - MQTT keep-alive (sending ping requests and processing ping responses) <br>
+ * The MQTT keep-alive mechanism involves sending pings to the server if the connection
+ * is idle. Therefore, in the absence of any other messages, <b>this function must be called
+ * at least once every keep-alive period to send the ping request</b>.
+ * - @ref mqtt_autoreconnect (if enabled) <br>
+ * If the client detects a disconnect, the reconnection will be performed in this function.
  *
- * @param pClient Reference to the IoT Client
- * @param timeout_ms Maximum number of milliseconds to pass thread execution to the client.
+ * @param[in] pClient MQTT client context
+ * @param[in] timeout_ms Amount of time to yield. This function will return to the caller
```

Comment:
Should this have any suggestion about the minimum amount of yield time? (It should be at least support the timeout of network read of incoming PUBLISH messages, followed b logic to determine sending ping request)

Reply:
Any suggestion would be dependent on the platform.  I don't think we can make a suggestion that works everywhere.